apps/full-site-editing/full-site-editing-plugin/site-editor-experiment/index.php

```diff
@@ -0,0 +1,162 @@
+		__( 'Site Editor (beta)', 'gutenberg' ),
+		'edit_theme_options',
+		'gutenberg-edit-site',
+		'gutenberg_edit_site_page',
+		'dashicons-edit'
+	);
+}
+
+/**
+ * Initialize the Gutenberg Edit Site Page.
+ * Copy of core function from 7.3.0 - gutenberg_edit_site_init
+ *
+ * Currently overriding to change hook requirement to allow Site Editor at top level sidebar.
+ * And to troubleshoot the wp.editSite.initialize CORS error.
+ *
+ * Any changes will be prefixed with @DOTCOM_CUSTOMIZATION comment.
+ *
+ * @param string $hook Page.
+ */
+function fse_plugin_edit_site_init( $hook ) {
+	global $_wp_current_template_id;
+	// @DOTCOM_CUSTOMIZATION - rename hook to allow toplevel
+	if ( 'toplevel_page_gutenberg-edit-site' !== $hook ) {
+		return;
+	}
+
+	// Get editor settings.
+	$max_upload_size = wp_max_upload_size();
+	if ( ! $max_upload_size ) {
+		$max_upload_size = 0;
+	}
+
+	// This filter is documented in wp-admin/includes/media.php.
+	$image_size_names      = apply_filters(
+		'image_size_names_choose',
+		array(
+			'thumbnail' => __( 'Thumbnail', 'gutenberg' ),
+			'medium'    => __( 'Medium', 'gutenberg' ),
+			'large'     => __( 'Large', 'gutenberg' ),
+			'full'      => __( 'Full Size', 'gutenberg' ),
+		)
+	);
+	$available_image_sizes = array();
+	foreach ( $image_size_names as $image_size_slug => $image_size_name ) {
+		$available_image_sizes[] = array(
+			'slug' => $image_size_slug,
+			'name' => $image_size_name,
+		);
+	}
+
+	$settings = array(
+		'disableCustomColors'    => get_theme_support( 'disable-custom-colors' ),
+		'disableCustomFontSizes' => get_theme_support( 'disable-custom-font-sizes' ),
+		'imageSizes'             => $available_image_sizes,
+		'isRTL'                  => is_rtl(),
+		'maxUploadFileSize'      => $max_upload_size,
+	);
+
+	list( $color_palette, ) = (array) get_theme_support( 'editor-color-palette' );
+	list( $font_sizes, )    = (array) get_theme_support( 'editor-font-sizes' );
+	if ( false !== $color_palette ) {
+		$settings['colors'] = $color_palette;
+	}
+	if ( false !== $font_sizes ) {
+		$settings['fontSizes'] = $font_sizes;
+	}
+
+	// Get root template by trigerring `./template-loader.php`'s logic.
+	get_front_page_template();
+	get_index_template();
+	apply_filters( 'template_include', null );
+	$settings['templateId'] = $_wp_current_template_id;
+
+	// Initialize editor.
+	wp_add_inline_script(
+		'wp-edit-site',
+		sprintf(
+			'wp.domReady( function() {
+				wp.editSite.initialize( "edit-site-editor", %s );
+			} );',
+			wp_json_encode( gutenberg_experiments_editor_settings( $settings ) )
+		)
+	);
+
+	// Preload server-registered block schemas.
+	wp_add_inline_script(
+		'wp-blocks',
+		'wp.blocks.unstable__bootstrapServerSideBlockDefinitions(' . wp_json_encode( get_block_editor_server_block_settings() ) . ');'
+	);
+
+	// @DOTCOM_CUSTOMIZATION - trigger enqueue_block_editor_assets to properly set apiFetch for dotcom etc.
+	do_action( 'enqueue_block_editor_assets' );
```

Comment:
2 from above - Trigger the enqueue_block_editor_assets action to allow things like apiFetch to be properly set up for dotcom.